Antidote Verifier/Sdl.Community.AntidoteVerifier/Sdl.Community.AntidoteVerifier/Utils/SegmentExtenstions.cs

```diff
@@ -5,110 +5,162 @@
 {
     public static class SegmentExtenstions
     {
         public static string GetString(this ISegment segment, bool includeSegments=false)
         {
-            var textVisitor = new TextCollectionVisitor(includeSegments);
+            var textVisitor = new CustomTextCollectionVisitor(segment);
 
             foreach (var item in segment)
             {
-                IStructureTag stag = item as IStructureTag;
-                if (stag != null) continue;
+                //IStructureTag stag = item as IStructureTag;
+                //if (stag != null) continue;
 
                 item.AcceptVisitor(textVisitor);
             }
 
             return textVisitor.CollectedText;
         }
 
-        public static IText GetTextAtLocation(this ISegment segment, int startIndex)
-        {
-            var counter = segment.GetCharacterCountingIterator(startIndex);
-            if (counter.CharacterCount < startIndex)
-            {
-                // the actual count is between this location and the next
-                //  - if this is a text node we can point to the exact location inside the text
-                IText text = counter.CurrentLocation.ItemAtLocation as IText;
-                return text;
-            }
-            return null;
-        }
+        //public static IText GetTextAtLocation(this ISegment segment, int startIndex)
+        //{
+        //		var counter = segment.GetCharacterCountingIterator(startIndex);
+        //		if (counter.CharacterCount < startIndex)
+        //		{
+        //				// the actual count is between this location and the next
+        //				//  - if this is a text node we can point to the exact location inside the text
+        //				IText text = counter.CurrentLocation.ItemAtLocation as IText;
+        //				return text;
+        //		}
+        //		return null;
+        //}
 
         public static string Substring(this ISegment segment, int startIndex, int endPosition)
         {
-            var counter = segment.GetCharacterCountingIterator(startIndex);
-            if (counter.CharacterCount < startIndex)
+            var segmentText = GetString(segment);
+            if (segmentText.Length >= endPosition)
             {
-                // the actual count is between this location and the next
-                //  - if this is a text node we can point to the exact location inside the text
-                IText text = counter.CurrentLocation.ItemAtLocation as IText;
-                if (text != null)
-                {
-                    var startLocationInsideTextItem = new TextLocation(counter.CurrentLocation, startIndex - counter.CharacterCount);
-                    var segmentText = text.Properties.Text;
-                    return segmentText.Substring(startLocationInsideTextItem.TextOffset,
-                        endPosition - startLocationInsideTextItem.TextOffset);
-
-                }
+                return segmentText.Substring(startIndex, endPosition - startIndex);
```

Comment:
Current implementation assumes that the text send from Studio to Antidote is without tags. If this is no longer the case than this is ok otherwise it will not work correct.